pkg/operator/controllers/muo/deploy.go

```diff
@@ -0,0 +1,96 @@
+package muo
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	appsv1 "k8s.io/api/apps/v1"
+	kruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+
+	"github.com/Azure/ARO-RP/pkg/util/dynamichelper"
+	"github.com/Azure/ARO-RP/pkg/util/ready"
+)
+
+type deployer struct {
+	kubernetescli kubernetes.Interface
+	dh            dynamichelper.Interface
+}
+
+func newDeployer(kubernetescli kubernetes.Interface, dh dynamichelper.Interface) *deployer {
+	return &deployer{
+		kubernetescli: kubernetescli,
+		dh:            dh,
+	}
+}
+
+func (o *deployer) resources(pullspec string) ([]kruntime.Object, error) {
+	results := []kruntime.Object{}
+	for _, assetName := range AssetNames() {
+		b, err := Asset(assetName)
+		if err != nil {
+			return nil, err
+		}
+
+		obj, _, err := scheme.Codecs.UniversalDeserializer().Decode(b, nil, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		// set the image for the deployments
+		if d, ok := obj.(*appsv1.Deployment); ok {
+			if d.Labels == nil {
+				d.Labels = map[string]string{}
+			}
+
+			for i := range d.Spec.Template.Spec.Containers {
+				d.Spec.Template.Spec.Containers[i].Image = pullspec
+			}
+		}
+
+		results = append(results, obj)
+	}
+
+	return results, nil
+}
+
+func (o *deployer) CreateOrUpdate(ctx context.Context, imagePullspec string) error {
+	resources, err := o.resources(imagePullspec)
+	if err != nil {
+		return err
+	}
+
+	err = dynamichelper.Prepare(resources)
+	if err != nil {
+		return err
+	}
+
+	err = o.dh.Ensure(ctx, resources...)
+	return err
+}
+
+func (o *deployer) Remove(ctx context.Context) (errs []error) {
+	resources, err := o.resources("")
+	if err != nil {
+		return []error{err}
+	}
+
+	for _, obj := range resources {
+		// delete any deployments we have
+		if d, ok := obj.(*appsv1.Deployment); ok {
```

Comment:
Why are we caring only about `deployment` cleanup in the Remove step? Shouldn't we also remove Roles/Bindings/Service Accounts created with the MUO manifest?

Reply:
Right now the priority is "have MUO not run", full cleanup and ordering is something complicated enough that I elided it for this RP release.

Reply:
yes, IMO It sounds like a good story for a follow-up epic.